### PR TITLE
feat(argo-rollouts): Add initContainers to controller pod, allow secrets to be manipulated

### PR DESCRIPTION
**Pull request by lukepatrick**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

**Comment by jmeridth** [comment]
@lukepatrick please fix conflict and I'll initiate the test run again

**Comment by lukepatrick** [comment]
@jmeridth fixed - thanks

File: charts/argo-rollouts/templates/controller/deployment.yaml [hunk 1]
```diff
@@ -65,6 +65,10 @@ spec:
       {{- with .Values.controller.extraContainers }}
         {{- toYaml . | nindent 6 }}
       {{- end }}
+      {{- if .Values.controller.initContainers }}
+      initContainers:
+      {{- toYaml .Values.controller.initContainers | nindent 8 }}
+      {{- end }}
```
Comment (mkilchhofer) [inline charts/argo-rollouts/templates/controller/deployment.yaml:71]: Let's use the newer syntax :D

```suggestion
      {{- with .Values.controller.initContainers }}
      initContainers:
      {{- toYaml . | nindent 8 }}
      {{- end }}
```

**Review by jmeridth — APPROVED** [review]
LGTM 👍

**Comment by jmeridth** [comment]
@lukepatrick can you elaborate on why secrets needed to be manipulated?  We're about to [merge a change](https://github.com/argoproj/argo-helm/pull/1774/files#diff-2552b67fe51dae7122d9de83a0005b84c118e6709f1afe8733c100f578f7b225L102-L104) that will remove those verbs to be at parity with upstream.

**Pull request merged**